### PR TITLE
Benjamin/floor plan desks availability

### PR DESCRIPTION
**Pull request by Benjamin-GitHub**
Show up name on booked desks + taken desks as "red" else "green".

### CODE REVIEW

File: client/src/pages/FloorPlan.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
-const Desk = ({ desk }) => {
+const Desk = ({ desk, bookingForThisDesk }) => {
 	return (
 		<>
 			<g transform={`rotate(${desk.r} 75 75)`}>
 				<rect
 					style={{
-						fill: "#3F94D5",
+						fill: bookingForThisDesk ? "red" : "green",
```
Comment (gregdyke) [inline client/src/pages/FloorPlan.js]: can you move this style to a stylesheet and use className here?

File: client/src/pages/FloorPlan.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import React from "react";
 
-const Desk = ({ desk }) => {
+const Desk = ({ desk, bookingForThisDesk }) => {
```
Comment (gregdyke) [inline client/src/pages/FloorPlan.js]: bookingForThisDesk is a great name! (I would have just used booking :D )

File: client/src/pages/FloorPlan.js [hunk 3]
```diff
@@ -22,11 +22,19 @@ const Desk = ({ desk }) => {
...
 			</text>
+			<text fontSize={25} x={20} y={140}>
+				{bookingForThisDesk?.name}
+			</text>
 		</>
 	);
 };
 
-const FloorPlan = ({ desks }) => {
+const FloorPlan = ({ desks, bookingsByDate }) => {
```
Comment (gregdyke) [inline client/src/pages/FloorPlan.js]: I think this should be just bookings. Then a FloorPlan can show any bookings it gets told to show, without worrying what they are (this is only a change to the variable name)

File: client/src/pages/FloorPlan.js [hunk 4]
```diff
@@ -22,11 +22,19 @@ const Desk = ({ desk }) => {
...
+	const bookingsByDesk = {};
+	bookingsByDate.forEach((booking) => {
+		bookingsByDesk[booking.desk_id] = booking;
```
Comment (gregdyke) [inline client/src/pages/FloorPlan.js:32]: magic!

File: client/src/pages/ModalBookingScreen.js [hunk 5]
```diff
@@ -17,7 +17,12 @@ import {
 	Dropdown,
 } from "semantic-ui-react";
 
-function ModalBookingScreen({ bookingDate, refreshBooking, desks }) {
+function ModalBookingScreen({
+	bookingDate,
+	refreshBooking,
+	desks,
+	bookingsByDate,
```
Comment (gregdyke) [inline client/src/pages/ModalBookingScreen.js]: I think here it should be `bookings`. Or `bookingsForDate`, not `bookingsByDate` - bookingsByDate is the name of a lookup object where you can put a date in and get an array of bookings

File: client/src/pages/WeeklyTable.js [hunk 6]
```diff
@@ -74,7 +74,7 @@ const WeeklyTable = ({ bookings, desks, refreshBooking, maxDesksForDay }) => {
 	}
 
 	const bookingsByRow = getBookingsByRow(bookings, week);
-	const bookingsByDesk = getBookingsByDesk(bookings, week, desks);
+	const {bookingsByDesk, bookingsByDayWithDesk} = getBookingsByDesk(bookings, week, desks);
```
Comment (gregdyke) [inline client/src/pages/WeeklyTable.js:83]: nice!!!!!

**Review by gregdyke — DISMISSED** [review]
React is so great :D (without react, this would have been the point where the code becomes an incomprehensible spaghetti mess. And even with React, this is a particularly nice change, doing so much with so little!)

File: client/src/pages/FloorPlan.js (hunk 1 above)
Comment (TheCheekyGit) [reply client/src/pages/FloorPlan.js]: looks good with a wee change to reflect the new colour scheme.

Comment (Benjamin-GitHub) [reply client/src/pages/FloorPlan.js]: @gregdyke Yes, I will.
@TheCheekyGit let's do it at the relevant PR. "Style-Modal"

File: client/src/pages/FloorPlan.js (hunk 2 above)
Comment (Benjamin-GitHub) [reply client/src/pages/FloorPlan.js]: let's use greater than greater than great

File: client/src/pages/ModalBookingScreen.js (hunk 5 above)
Comment (Benjamin-GitHub) [reply client/src/pages/ModalBookingScreen.js]: I see!

**Review by viktoriia-kosenko — APPROVED** [review]

**Pull request merged**